test/e2e/resize_nodes.go

```diff
@@ -0,0 +1,331 @@
+		ObjectMeta: api.ObjectMeta{
+			Name: name,
+		},
+		Spec: api.ReplicationControllerSpec{
+			Replicas: replicas,
+			Selector: map[string]string{
+				"name": name,
+			},
+			Template: &api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: map[string]string{"name": name},
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  name,
+							Image: image,
+							Ports: []api.ContainerPort{{ContainerPort: 9376}},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createServeHostnameReplicationController(c *client.Client, ns, name string, replicas int) *api.ReplicationController {
+	By(fmt.Sprintf("creating replication controller %s", name))
+	rc, err := c.ReplicationControllers(ns).Create(newReplicationController(name, replicas, "gcr.io/google_containers/serve_hostname:1.1"))
+	Expect(err).NotTo(HaveOccurred())
+	return rc
+}
+
+func resizeReplicationController(c *client.Client, ns, name string, replicas int) {
+	rc, err := c.ReplicationControllers(ns).Get(name)
+	Expect(err).NotTo(HaveOccurred())
+	rc.Spec.Replicas = replicas
+	_, err = c.ReplicationControllers(rc.Namespace).Update(rc)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func waitForPodsCreated(c *client.Client, ns, name string, replicas int) *api.PodList {
+	// List the pods, making sure we observe all the replicas.
+	listTimeout := time.Minute
+	listInterval := 5 * time.Second
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
+	var pods *api.PodList
+	t := time.Now()
+	for {
+		var err error
+		pods, err = c.Pods(ns).List(label, fields.Everything())
+		Expect(err).NotTo(HaveOccurred())
+
+		Logf("Controller %s: Found %d pods out of %d", name, len(pods.Items), replicas)
+		if len(pods.Items) == replicas {
+			break
+		}
+		if time.Since(t) > listTimeout {
+			Failf("Controller %s: Gave up waiting for %d pods to come up after seeing only %d pods after %v seconds",
+				name, replicas, len(pods.Items), time.Since(t).Seconds())
+		}
+		time.Sleep(listInterval)
+	}
+	return pods
+}
+
+func waitForPodsRunning(c *client.Client, pods *api.PodList) {
+	// Wait for the pods to enter the running state. Waiting loops until the pods
+	// are running so non-running pods cause a timeout for this test.
+	By("ensuring each pod is running")
+	for _, pod := range pods.Items {
+		err := waitForPodRunningInNamespace(c, pod.Name, pod.Namespace)
+		Expect(err).NotTo(HaveOccurred())
+	}
+}
+
+func verifyPodsListining(c *client.Client, ns, name string, pods *api.PodList) {
+	By("trying to dial each unique pod")
+	retryTimeout := 2 * time.Minute
+	retryInterval := 5 * time.Second
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
+	err := wait.Poll(retryInterval, retryTimeout, podResponseChecker{c, ns, label, name, pods}.checkAllResponses)
+	if err != nil {
+		Failf("Did not get expected responses within the timeout period of %.2f seconds: %v", retryTimeout.Seconds(), err)
```

Comment:
I see this is done above, but I think the time printing can be done by just using `%v` and passing `retryTimeout` directly, rather than converting to seconds and using `%.2f seconds` annotation.